build: Migrate from Webpack to Vite (closes #146, resolves #238). (#175)
y-scope/yscope-log-viewer: docs/src/dev-guide/coding-guidelines.md

This project adheres to YScope's [contribution guidelines][yscope-guidelines] as well as the
project-specific guidelines below.

# Web Workers

## Importing Web Workers

> When importing Web Worker files, use Vite's `?worker` query suffix syntax:

Reviewer: I don't think "Web Workers" needs to be treated as a proper noun. I know MDN does so, but I can't see why they would. So can we capitalize is like normal words instead?

Author: Right, even in the specs https://html.spec.whatwg.org/multipage/workers.html they don't write the words in Title Case. let me update the docs

```ts
import MainWorker from "../services/MainWorker.worker?worker";

const worker = new MainWorker();
```

This special syntax tells Vite to transform the import as a Web Worker constructor. See
[Vite's Web Worker documentation][vite-worker-query-suffix] for more details.

# Naming

## Web Worker files

> ### File naming convention
> 
> Name Web Worker files with the `.worker.ts` extension. This convention:
> * Follows standard practices.
> * Works with [`eslint.config.mjs`][eslint-config-mjs] which ignores `.worker.ts` files, suppressing

Reviewer: ```suggestion
Name web worker files with the extension, `.worker.ts`. This is to:

* follow standard practices.
* allow [eslint.config.mjs][eslint-config-mjs] to ignore `.worker.ts` files, suppressing
```

* `arrayIndexIdx` for a 0-based indexing variable.
* `numEvents` for the total number of events.

## Web worker files

> Name worker files with a `.worker.ts` extension. Import them using Vite's `?worker` syntax:

Author: > If I'm not mistaken, the first guideline is really just so that the worker files are ignored by the import/default linting rule?

Partially. We used the `.worker.ts` filename extension because I came to realize that is a common practice. Then it happens that there are ESLint errors that cannot be easily resolved because the ESLint plugin does not know about the transformation of workers in Vite (due to the way we import) during compile time. Anyways, all those are worth documenting.

> Name worker files with a `.worker.ts` extension. Import them using Vite's `?worker` syntax:

Author: The split is done but the import syntax description no longer fits in the "Naming" section. I created a Web Worker section to hold it instead. Let me know if the flow doesn't make sense.
